pkg/cmd/admin/groups/sync/rfc2307/ldapinterface.go

```diff
@@ -57,6 +59,9 @@ type LDAPInterface struct {
 	// cachedUsers holds the result of user queries for later reference, indexed on user UID
 	// e.g. this will map an LDAP user UID to the LDAP entry returned from the query made using it
 	cachedUsers map[string]*ldap.Entry
+
+	// Err is used to log errors found interfacing with LDAP
+	Err io.Writer
```

Comment:
> having the check of an outofbounds error log and continue could be part of the default error handler implementation

@stevekuznetsov There are couple ways to build these, but I think he means something like this:

```go
type ErrorHandler interface{
    // HandleError gets to process an error and decide whether or not it is fatal.  Processing
    // should not mutate the error, but may do things like display errors or fire off events.
    // Fatal errors can be indicated by returning back a non-nil error.
    // context contains data relevant to the error
    // handled indicates that the handler did something with the error.  This can be used
    // to easily chain a priority of handlers to handle specific conditions
    HandleError(err error, context interface{}) (handled bool, err error)
}

type MemberLookupErrorHandler struct{
    Err io.Writer
}

func (e *MemberLookupErrorHandler) HandleError(err error, context interface{}) (bool, error){
    if ldapMemberUID, ok := context.(string); ldaputil.IsQueryOutOfBoundsError(err) && ldapMemberUID{
        // do something
        // return true, nil
    }

    return false, nil
}
```